packages/core/test/invoke.test.ts

```diff
@@ -2709,6 +2709,44 @@ describe('invoke', () => {
       interpret(machine).start();
     });
   });
+
+  it('invoke generated ID should be predictable based on the state node where it is defined', (done) => {
```

Comment:
this is an OK test but doesn't directly test the bug that is being fixed 😉 a regression test can be created based on the code included in the original report [here](https://github.com/statelyai/xstate/issues/464)

Reply:
It is not easy to make a repro from the original code, since it depends on non-working parallel state behavior (`onDone` isn't called on a top-level parallel state).

Reply:
> It is not easy to make a repro from the original code, since it depends on non-working parallel state behavior (onDone isn't called on a top-level parallel state).

Apparently, this worked 2.5 years ago - the old viz still call that `onDone` with the very exact code from the issue: https://xstate.js.org/viz/?gist=812dc699e7161ba31887af2d19e7b997 . It also loops infinitely for some reason because both services (in parallel states) are constantly being restarted~., oops 🤫

Reply:
I've tweaked that test case [here](https://github.com/statelyai/xstate/pull/2909) and rechecked with the `main` branch - it fails there.